power.inc.php

```diff
@@ -1430,18 +1439,35 @@ function getPanelListBySource( $onlyFirstGen = false ) {
 		while ( $row = $st->fetch() ) {
 			$pList[] = $row;
 		}
-		
 		if ( ! $onlyFirstGen ) {
 			foreach ( $pList as $currPan ) {
-				$st->execute( array( ":ParentPanelID"=>$currPan->PanelID ) );
-				while ( $row = $st->fetch() ) {
-					$pList[] = $row;
-				}
+				$this->ParentPanelID=$currPan->PanelID;
+				$pList = array_merge($pList, $this->getPanelListBySource());
 			}
 		}
 		
 		return $pList;
 	}
+
+	function getSubPanels( $onlyFirstGen = false ) {
+		$sql = "select * from fac_PowerPanel where ParentPanelID=:ParentPanelID order by PanelLabel ASC";
+		$st = $this->prepare($sql);
+		$st->setFetchMode(PDO::FETCH_CLASS, "PowerPanel");
+		$st->execute(array(":ParentPanelID"=>$this->PanelID));
```

Comment:
This could be accomplished with $pp=new PowerPanel();  $pp->ParentPanelID=$this->PanelID;  $pp->Search();